src/Mario.purs

```diff
@@ -62,7 +65,7 @@ applyGravity c | c.y <= -c.dy = c { y = 0.0, dy = 0.0 }
 applyGravity c = c { dy = c.dy - gravity }
 
 -- Mario can move himself left/right with a fixed acceleration
-walk :: Boolean -> Boolean -> Character -> Character
+walk :: Partial => Boolean -> Boolean -> Character -> Character
```

Comment:
This isn't actually partial, so no need for the partial constraint. Instead, we can remove the unnecessary final guard in `applyFriction`. I only wrote it there for pedagogical purposes.

```diff
-   applyFriction c | c.dx < 0.0 = c { dx = c.dx + friction c }
+   applyFriction c { dx = c.dx + friction c }
```